android/shopify_buy_plugin/src/main/java/com/shopify/unity/buy/models/PricingLineItems.java

```diff
@@ -4,35 +4,70 @@
+    private static final String SUBTOTAL = "subtotal";
+    private static final String TAX_PRICE = "taxPrice";
+    private static final String TOTAL_PRICE = "totalPrice";
+    private static final String SHIPPING_PRICE = "shippingPrice";
+
     @NonNull public final BigDecimal subtotal;
     @NonNull public final BigDecimal taxPrice;
     @NonNull public final BigDecimal totalPrice;
 
     @Nullable public final BigDecimal shippingPrice;
 
-    private static final JsonAdapter<PricingLineItems> jsonAdapter =
-            new Moshi.Builder().add(new BigDecimalAdapter()).build().adapter(PricingLineItems.class);
-
-    private PricingLineItems() {
-        this.totalPrice = BigDecimal.ZERO;
-        this.subtotal = BigDecimal.ZERO;
-        this.taxPrice = BigDecimal.ZERO;
-        this.shippingPrice = null;
+    private PricingLineItems(@NonNull BigDecimal subtotal,
+                             @NonNull BigDecimal taxPrice,
+                             @NonNull BigDecimal totalPrice,
+                             @Nullable BigDecimal shippingPrice) {
+        this.totalPrice = totalPrice;
+        this.subtotal = subtotal;
+        this.taxPrice = taxPrice;
+        this.shippingPrice = shippingPrice;
     }
 
-    public static PricingLineItems fromJsonString(String jsonString) throws IOException {
-        return jsonAdapter.fromJson(jsonString);
+    public static PricingLineItems fromJsonString(String jsonString) throws JSONException {
+        JSONObject json = new JSONObject(jsonString);
+        BigDecimal subtotal = decimalPropertyFromJson(json, SUBTOTAL);
+        BigDecimal taxPrice = decimalPropertyFromJson(json, TAX_PRICE);
+        BigDecimal totalPrice = decimalPropertyFromJson(json, TOTAL_PRICE);
+
+        BigDecimal shippingPrice = null;
+        if (json.has(SHIPPING_PRICE)) {
+             shippingPrice = nullableDecimalPropertyFromJson(json, SHIPPING_PRICE);
+        }
+
+        return new PricingLineItems(subtotal, taxPrice, totalPrice, shippingPrice);
     }
 
     @Override
-    public String toJsonString() {
-        return jsonAdapter.toJson(this);
+    public String toJsonString() throws JSONException {
+        JSONObject json = new JSONObject();
+        json.put(SUBTOTAL, subtotal.toString());
+        json.put(TAX_PRICE, taxPrice.toString());
+        json.put(TOTAL_PRICE, totalPrice.toString());
+
+        if (shippingPrice != null) {
+            json.put(SHIPPING_PRICE, shippingPrice.doubleValue());
+        }
+
+        return json.toString();
+    }
+
+    private static BigDecimal nullableDecimalPropertyFromJson(JSONObject obj, String property) throws JSONException {
+        if (obj.has(property)) {
+            return decimalPropertyFromJson(obj, property);
+        }
+        return null;
+    }
+
+    private static BigDecimal decimalPropertyFromJson(JSONObject obj, String property) throws JSONException {
+        return new BigDecimal(obj.getString(property)).setScale(2, BigDecimal.ROUND_FLOOR);
```

Comment:
Note: All of these Moshi changes are in this PR https://github.com/Shopify/unity-buy-sdk/pull/318